src/components/molecules/LoadingPage/LoadingPage.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useMemo, useState } from "react";
+
+import { CustomLoader } from "types/CustomLoader";
+
+import { useCustomLoaders } from "hooks/useCustomLoaders";
+
+import "./loading.scss";
+
+const getRandomInt = (max: number, randomValue: () => number = Math.random) => {
+  return Math.floor(randomValue() * Math.floor(max));
+};
+
+// TODO: move this into useCustomLoaders, or settings, or similar?
+export const DEFAULT_LOADER: CustomLoader = {
+  title: "Loading Sparkle",
+  text: "Loading Sparkle",
+  url: "",
+};
+
+export const LoadingPage = () => {
+  const { customLoaders } = useCustomLoaders();
+
+  // TODO: randomly choose one of the custom loaders to use
+  //   This should be 'stable per load' and not change even if there are multiple re-renders
+  // TODO: for some reason it seems to run at least twice, with changing int values.. which makes it 'flash' and change backgrounds
+  // TODO: move all of this into useCustomLoaders or similar?
```

Comment:
[TODO]

```typescript
// TODO: randomly choose one of the custom loaders to use
//   This should be 'stable per load' and not change even if there are multiple re-renders
// TODO: for some reason it seems to run at least twice, with changing int values.. which makes it 'flash' and change backgrounds
// TODO: move all of this into useCustomLoaders or similar?
```

Reply:
> for some reason it seems to run at least twice, with changing int values.. which makes it 'flash' and change backgrounds

For this part, it seems the 'issue' is not in my code/implementation, but more in the way the app is structured.

- `src/index.tsx` renders `LoadingPage` as part of the `AuthIsLoaded` component
- `src/pages/VenuePage/VenuePage.tsx` renders `LoadingPage` while waiting for the venue data to be fetched.

Since both of these happen during 'first app load', that's why we see this 'double loading' 'flash'.

Since I think the only real time we actually see the loading screen at the moment is on a 'new page load' (eg. when we navigate without using the routing helpers), I think we could hack around this by storing the 'stable random integer' in the context itself. It wouldn't re-choose the random loading screen to choose 'within the same page load', but based on my theory above, that probably doesn't matter right now.

Reply:
fixed in [`327b240` (#1240)](https://github.com/sparkletown/sparkle/pull/1240/commits/327b240f79d95ae502556f51f4c7dd4920cf3dd4) + [`73cd376` (#1240)](https://github.com/sparkletown/sparkle/pull/1240/commits/73cd37612f3c354d6f1f6ff2cdeef23707f0bc3a) + [`c47d191` (#1240)](https://github.com/sparkletown/sparkle/pull/1240/commits/c47d19134f03010c466fd6ba163198c2e91de9fc)